### ISSUE TITLE
Unable to reproduce the effect of the paper

### ISSUE THREAD

**Issue by lucienJxy** [Haiyang-W/CAGroup3D#23]
Thanks to the great work of you and your team, I am trying to replicate your algorithm. I use a single 3090 GPU for training. According to the code you provided and the steps you gave, the eval score of the model I trained is much different from the value you provided in the paper.I got  'mAP_0.25': 0.5806578397750854, 'mAP_0.50': 0.3567165732383728. Would you like to tell me the reason? Is it due to too few experiments? Do I need to modify some parameters in the configuration file? You mentioned that the data in the paper is based on mmdet3d. Is the low score related to the framework of the algorithm? Is it convenient for you to provide the code of mmdet3d? Thank you very much for your answer and help.
2023-09-06 11:04:27,091   INFO  *************** EPOCH 14 EVALUATION *****************
2023-09-06 11:08:26,943   INFO  *************** Performance of EPOCH 14 *****************
2023-09-06 11:08:26,944   INFO  Generate label finished(sec_per_example: 0.0475 second).
2023-09-06 11:08:26,944   INFO  recall_roi_0.25: 0.000000
2023-09-06 11:08:26,944   INFO  recall_rcnn_0.25: 0.000000
2023-09-06 11:08:26,944   INFO  recall_roi_0.5: 0.000000
2023-09-06 11:08:26,945   INFO  recall_rcnn_0.5: 0.000000
2023-09-06 11:08:26,955   INFO  Average predicted number of objects(5050 samples): 135.440
2023-09-06 11:08:47,886   INFO  {'bed_AP_0.25': 0.8639285564422607, 'table_AP_0.25': 0.5081688761711121, 'sofa_AP_0.25': 0.6558625102043152, 'chair_AP_0.25': 0.8160231709480286, 'toilet_AP_0.25': 0.9042611122131348, 'desk_AP_0.25': 0.25037357211112976, 'dresser_AP_0.25': 0.2574673593044281, 'night_stand_AP_0.25': 0.6470746397972107, 'bookshelf_AP_0.25': 0.14564046263694763, 'bathtub_AP_0.25': 0.757777988910675, 'mAP_0.25': 0.5806578397750854, 'bed_rec_0.25': 0.9689320388349515, 'table_rec_0.25': 0.9356899488926746, 'sofa_rec_0.25': 0.9521531100478469, 'chair_rec_0.25': 0.9515774760383386, 'toilet_rec_0.25': 0.9862068965517241, 'desk_rec_0.25': 0.8921360255047821, 'dresser_rec_0.25': 0.8807339449541285, 'night_stand_rec_0.25': 0.9333333333333333, 'bookshelf_rec_0.25': 0.6985815602836879, 'bathtub_rec_0.25': 0.9183673469387755, 'mAR_0.25': 0.9117711681380245, 'bed_AP_0.50': 0.46258190274238586, 'table_AP_0.50': 0.266003280878067, 'sofa_AP_0.50': 0.4439016580581665, 'chair_AP_0.50': 0.6562656760215759, 'toilet_AP_0.50': 0.6436388492584229, 'desk_AP_0.50': 0.051447901874780655, 'dresser_AP_0.50': 0.1490955799818039, 'night_stand_AP_0.50': 0.5124972462654114, 'bookshelf_AP_0.50': 0.013978171162307262, 'bathtub_AP_0.50': 0.36775514483451843, 'mAP_0.50': 0.3567165732383728, 'bed_rec_0.50': 0.566990291262136, 'table_rec_0.50': 0.528534923339012, 'sofa_rec_0.50': 0.6762360446570973, 'chair_rec_0.50': 0.7881389776357828, 'toilet_rec_0.50': 0.7517241379310344, 'desk_rec_0.50': 0.35812964930924546, 'dresser_rec_0.50': 0.481651376146789, 'night_stand_rec_0.50': 0.7137254901960784, 'bookshelf_rec_0.50': 0.18085106382978725, 'bathtub_rec_0.50': 0.5510204081632653, 'mAR_0.50': 0.5597002362470228}
2023-09-06 11:08:47,888   INFO  Result is save to /root/autodl-tmp/CAGroup3D/output/sunrgbd_models/CAGroup3D/default/eval/epoch_14/val/default
2023-09-06 11:08:47,888   INFO  ****************Evaluation done.*****************

**Comment by Haiyang-W** [comment]
Did you use the data we provided?  We use the data generated by mmdet3d 0.15, which is different from the current mmdet (>1.0) ( some difference about the Coordinate system). Please use the data we provided, or generate it by mmdet3d 0.15.

Your performance is so low, so there must be something wrong with the training. You can download our pre-trained model and inference first.

**Comment by Haiyang-W** [comment]
On our side, sunrgbd can easily reproduce to the reported performance.

**Comment by Haiyang-W** [comment]
Notably, we train our model on 4 RTX 3090 GPU cards with batch size = 4 of each card(total batch size is 16). A single gpu may have some problems in batch_size.

**Comment by lucienJxy** [comment]
> Fine,I will check the data which I am using and evaluae the  pre-trained model first.Thank you for your qulick answer,wish you best.

**Comment by Haiyang-W** [comment]
Wish you all the best.

**Comment by lucienJxy** [comment]
Hello, I replaced the data you provided and tested your pre-trained model, and got 'mAP_0.25': 0.6710548400878906, 'mAP_0.50': 0.49219608306884766. After that,I carried out another training with a single 3090 gpu and got 'mAP_0.25': 0.6418028473854065, mAP_0.50': 0.4689980447292328,which still lower than you reported 66.8 and 50.2. But I do not think that's a problem, as you mentioned ,a single gpu may have some problems in batch_size. By the way,I found that  the data we provided only have points and two 'pkl' files,I want to know how to get the labels and other infos such as images if I want to make a fusion of different features?To put it another way, where can the label data(gt_boxes?) be found  and how does the algorithm import it.Because the data I used before was preprocessed according to mmdetection3d, it was different from the data you provided. Perhaps I have overlooked some details. I hope to get your kindly reply. Thank you very much

**Comment by Haiyang-W** [comment]
Could you use the 4 GPUs with batch_size = 4 of each card? or set the total batch size to 16, so that the number of training iters is the same as ours. Or reduce the learning rate according to your batch size. If you directly run the train config with single GPUs, it will drop 2~3 points in my experiments.

**Comment by Haiyang-W** [comment]
We preprocess the dataset based on mmdetection3d 0.15, you can reprocess the original dataset by mmdet3d(0.15).

**Comment by Haiyang-W** [comment]
I think it's common sense that matches the batch size and learning rate. Few people train sunrgbd by a single card, generally two cards or 4 cards, so that the batch size can be 16. 

We can easily reproduce the results in our side, if you can accurately follow our training strategy, I believe it can be easily replicated. (4 gpus x 4  samples or 2 gpus x 8 samples for each iter).

**Comment by Haiyang-W** [comment]
We provide training logs, you can refer to align our training logs to find problems.
